### ISSUE TITLE
Polygon Boolean Operation - !outer_ccbs_to_remove.empty()

### ISSUE THREAD

**Issue by bryantcurto** [CGAL/cgal#3989]
Hello all!

## Details

I've run into an issue while performing boolean difference (using the EPEC Kernel). The assertion failure is:

```
terminate called after throwing an instance of 'CGAL::Assertion_exception' 
  what():  CGAL ERROR: assertion violation! 
Expr: !outer_ccbs_to_remove.empty() 
File: 
[PATH_TO_CGAL]/include/CGAL/Boolean_set_operations_2/Gps_on_surface_base_2.h 
Line: 1254 
Aborted (core dumped)
```

Disabling assertions, this issue results in a Segmentation fault.

## Minimal Working Example
```
    Polygon_2<K> ob;
    ob.push_back(Point_2<K>(1, 1)); 
    ob.push_back(Point_2<K>(1, 0)); 
    ob.push_back(Point_2<K>(6, 0)); 
    ob.push_back(Point_2<K>(6, 7)); 
    ob.push_back(Point_2<K>(0, 7)); 
    ob.push_back(Point_2<K>(0, 1)); 

    Polygon_2<K> h; 
    h.push_back(Point_2<K>(2, 1));
    h.push_back(Point_2<K>(2, 2));
    h.push_back(Point_2<K>(3, 2));
    h.push_back(Point_2<K>(3, 3));
    h.push_back(Point_2<K>(2, 3));
    h.push_back(Point_2<K>(2, 4));
    h.push_back(Point_2<K>(3, 4));
    h.push_back(Point_2<K>(3, 5));
    h.push_back(Point_2<K>(4, 5));
    h.push_back(Point_2<K>(4, 1));

    Polygon_with_holes_2<K> ob_with_holes(ob); 
    ob_with_holes.add_hole(h); 
    Polygon_set_2<K> inter(ob_with_holes); 

    Polygon_2<K> new_poly; 
    new_poly.push_back(Point_2<K>(1, 1));
    new_poly.push_back(Point_2<K>(2, 1));
    new_poly.push_back(Point_2<K>(2, 2));
    new_poly.push_back(Point_2<K>(2, 3));
    new_poly.push_back(Point_2<K>(2, 4));
    new_poly.push_back(Point_2<K>(2, 5));
    new_poly.push_back(Point_2<K>(3, 5));
    new_poly.push_back(Point_2<K>(4, 5));
    new_poly.push_back(Point_2<K>(4, 6));
    new_poly.push_back(Point_2<K>(1, 6));

    inter.difference(new_poly);
```

## Environment

* Operating system: Linux
* Compiler: gcc
* CGAL version:4.13.1
* Boost version: 1.67

I have also made an issue about this on the [dev forum](http://cgal-discuss.949826.n4.nabble.com/Polygon-Boolean-Operation-outer-ccbs-to-remove-empty-td4664203.html).

Edit: Made MWE more minimal.

**Comment by bryantcurto** [comment]
Playing around with the MWE, I find the following:
```
(ob - h) - new_poly ==> CRASH
(ob - new_poly) - h ==> CRASH
ob - (h + new_poly) ==> No Crash
ob - (new_poly + h) ==> No Crash
```

**Comment by sloriot** [comment]
I confirm the issue. Working on it.

**Comment by sloriot** [comment]
https://github.com/CGAL/cgal/pull/3998 fixes the issue for me.

**Comment by lrineau** [comment]
Fixed by merge of #3998.
